src/server/dflycmd.cc

```diff
@@ -651,6 +658,37 @@ void DflyCmd::StopReplication(uint32_t sync_id) {
   replica_infos_.erase(sync_id);
 }
 
+void DflyCmd::BreakStalledFlowsInShard() {
+  unique_lock lk(mu_, try_to_lock);
+  if (!lk.owns_lock())
+    return;  // give up
+
+  ShardId sid = EngineShard::tlocal()->shard_id();
+  vector<uint32_t> deleted;
+
+  for (auto [sync_id, replica_ptr] : replica_infos_) {
+    shared_lock lock = replica_ptr->GetSharedLock();
+
+    if (!replica_ptr->flows[sid].saver)
```

Comment:
if it's not in full sync phase - skip

Reply:
Add a TODO to handle stable sync in the future?